### ISSUE TITLE
Behavior of subclassing groups differs between `net` and `cell`.

### ISSUE THREAD

**Issue by michaeldeistler** [jaxleyverse/jaxley#338]
If one subclasses groups of `net` then one needs to use the **global** index of the cell (i.e. in the entire network):
```python
net = jx.Network([cell for _ in range(5)])
net.cell(3).add_to_group("exc")
net.exc.cell(3).set(...)
```

Contrary to that, for `cell`, one has to use the local index of the branch within the group:
```python
cell = jx.Cell(Branch, [-1, 0, 0])
cell.branch(2).add_to_group("exc")
cell.exc.branch(0).set(...)
```

**Comment by michaeldeistler** [comment]
I think this can be closed, right @jnsbck ?

**Comment by jnsbck** [comment]
Should have been addressed in #447, will add a test for this in #453
